Tasks/AzureCLIV3/azureclitask.ts

```diff
@@ -0,0 +1,463 @@
+            }
+            else {
+                Utility.throwIfError(tl.execSync("az", "login --identity --output none"), tl.loc("MSILoginFailed"));
+            }            
+        }
+        else {
+            throw tl.loc('AuthSchemeNotSupported', authScheme);
+        }
+
+        this.isLoggedIn = true;
+        if (!!subscriptionID) {
+            //set the subscription imported to the current subscription
+            Utility.throwIfError(tl.execSync("az", "account set --subscription \"" + subscriptionID + "\""), tl.loc("ErrorInSettingUpSubscription"));
+        }
+    }
+
+    private static async setupAzureDevOpsCLI(connectedService: string): Promise<void> {
+        try {
+            var authScheme: string = tl.getEndpointAuthorizationScheme(connectedService, true);
+            var visibleAzLogin: boolean = tl.getBoolInput("visibleAzLogin", true);
+             // Install Azure DevOps extension
+            Utility.throwIfError(tl.execSync("az", "extension add -n azure-devops -y"), tl.loc("FailedToInstallAzureDevOpsCLI"));
+
+            if (authScheme.toLowerCase() == "workloadidentityfederation") {
+                var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
+                var tenantId: string = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
+
+                const federatedToken = await this.getIdToken(connectedService);
+                tl.setSecret(federatedToken);
+                let args = `login --service-principal -u "${servicePrincipalId}" --tenant "${tenantId}" --allow-no-subscriptions --federated-token "${federatedToken}"`;
+
+                if(!visibleAzLogin ){
+                    args += ` --output none`;
+                }
+                //login using OpenID Connect federation
+                Utility.throwIfError(tl.execSync("az", args), tl.loc("LoginFailed"));
+
+                this.servicePrincipalId = servicePrincipalId;
+                this.federatedToken = federatedToken;
+                this.tenantId = tenantId;
+
+                process.env.AZURESUBSCRIPTION_SERVICE_CONNECTION_ID = connectedService;
+                process.env.AZURESUBSCRIPTION_CLIENT_ID = servicePrincipalId;
+                process.env.AZURESUBSCRIPTION_TENANT_ID = tenantId;
+
+                const organization = tl.getVariable('System.CollectionUri');
+                const project = tl.getVariable('System.TeamProject');
+
+                if (organization) {
+                    Utility.throwIfError(tl.execSync("az", `devops configure --defaults organization="${organization}"`), tl.loc("FailedToSetAzureDevOpsOrganization"));
+                }
+
+                if (project) {
+                    Utility.throwIfError(tl.execSync("az", `devops configure --defaults project="${project}"`), tl.loc("FailedToSetAzureDevOpsProject"));
+                }
+
+        }
```

Comment:
Please fix the indentation here for closing curly bracet.
Also I recommend throwing an error here if authScheme.toLowerCase() != "workloadidentityfederation"